$OpenBSD: patch-jdk_src_solaris_native_java_lang_java_props_md_c,v 1.4 2012/10/24 18:54:03 kurt Exp $

OpenBSD supports only the "C" and "POSIX" locales for all but the
LC_CTYPE locale.

--- jdk/src/solaris/native/java/lang/java_props_md.c.orig	Sat Aug  4 21:43:10 2012
+++ jdk/src/solaris/native/java/lang/java_props_md.c	Sat Aug  4 22:49:04 2012
@@ -142,7 +142,7 @@ static int ParseLocale(int cat, char ** std_language, 
     /* Query the locale set for the category */
     lc = setlocale(cat, NULL);
 
-#ifndef __linux__
+#if !defined(__linux__) && !defined(_ALLBSD_SOURCE)
     if (lc == NULL) {
         return 0;
     }
@@ -430,12 +430,19 @@ GetJavaProperties(JNIEnv *env)
                     &(sprops.format_country),
                     &(sprops.format_variant),
                     &(sprops.encoding))) {
+#ifdef __OpenBSD__
+        sprops.language = sprops.format_language;
+        sprops.script = sprops.format_script;
+        sprops.country = sprops.format_country;
+        sprops.variant = sprops.format_variant;
+#else
         ParseLocale(LC_MESSAGES,
                     &(sprops.language),
                     &(sprops.script),
                     &(sprops.country),
                     &(sprops.variant),
                     NULL);
+#endif
     } else {
         sprops.language = "en";
         sprops.encoding = "ISO8859-1";
